src/main/drivers/rcdevice.h

```diff
@@ -0,0 +1,226 @@
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED8                           = 8,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED9                           = 9,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED10                          = 10,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED11                          = 11,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED12                          = 12,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED13                          = 13,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED14                          = 14,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED15                          = 15,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED16                          = 16,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED17                          = 17,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED18                          = 18,
+    RCDEVICE_PROTOCOL_SETTINGID_RESERVED19                          = 19,
+} rcdeviceReservedSettingID_e;
+
+typedef enum {
+    RCDEVICE_PROTOCOL_SETTINGTYPE_UINT8                             = 0,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_INT8                              = 1,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_UINT16                            = 2,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_INT16                             = 3,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_FLOAT                             = 8,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_TEXT_SELECTION                    = 9,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_STRING                            = 10,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_FOLDER                            = 11,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_INFO                              = 12,
+    RCDEVICE_PROTOCOL_SETTINGTYPE_UNKNOWN
+} rcdeviceSettingType_e;
+// end of Runcam Device definition
+
+// Old version defination(RCSplit firmware v1.0.0 and v1.1.0)
+// packet header and tail
+#define RCSPLIT_PACKET_HEADER                                       0x55
+#define RCSPLIT_PACKET_CMD_CTRL                                     0x01
+#define RCSPLIT_PACKET_TAIL                                         0xaa
+
+typedef enum {
+    RCSPLIT_CTRL_ARGU_INVALID                                       = 0x0,
+    RCSPLIT_CTRL_ARGU_WIFI_BTN                                      = 0x1,
+    RCSPLIT_CTRL_ARGU_POWER_BTN                                     = 0x2,
+    RCSPLIT_CTRL_ARGU_CHANGE_MODE                                   = 0x3,
+    RCSPLIT_CTRL_ARGU_WHO_ARE_YOU                                   = 0xFF,
+} rcsplit_ctrl_argument_e;
+// end of old version protocol definition
+
+typedef struct {
+    rcdevice_protocol_version_e protocolVersion;
+    uint16_t features;
+} runcamDeviceInfo_t;
+
+typedef struct _runcamDeviceSetting {
+    uint8_t id;
+    char *name;
+    char *value;
+    struct _runcamDeviceSetting *next;
+} runcamDeviceSetting_t;
+
+typedef struct _runcamDeviceSettingTextSelection {
+    char *text;
+    struct _runcamDeviceSettingTextSelection *next;
+} runcamDeviceSettingTextSelection_t;
+
+typedef struct {
```

Comment:
Betaflight style is for all `typedef struct`s to have a name, so this should be `typedef struct runcamDeviceSettingDetail_s` and so on for all others in this file.